### ISSUE TITLE
[Improvement] Really Disable system messages in chats and rooms

### ISSUE THREAD

**Issue by ankar84** [RocketChat/Rocket.Chat#16931]
### Description:

New method #16243 of hide system messages is not so good for big read-only channels (maybe default channels).
First time when #4760 was implemented hide options works much better then now.
System messaged was really disabled but not just hide.
My idea - for private groups system message (e.g. join or leave messages) may be very important for security reasons - who was added by whom.
But for read-only channels (which usually public and used for news and announcements) system messages is like spam (not so important information).  If you add 5000 users to public read-only channel for announcements you really will see 5000 messages in that channel (even if there is no regular, not system messages)
Previously (before #16243 and after #4760) when read-only channel owner set **hide to join system messages** and 5000 users will join that channel after that - there will be 0 messages in that channel (in **Admin UI** => **Rooms**), which is good, IMHO.

### Steps to reproduce:

1. Hide join channel system message for read-only channel
2. Add 100 users to that read-only channel
3. In **Admin UI** => **Rooms** you can see 100 messages in that Channel

### Expected behavior:

**Really Disable** system messages in read-only channels like it was before #16243

### Actual behavior:

System messages operate as regular messages, but in read-only channels it's like spam.
Rocket.Chat.ReactNative version 4.2 do not use new hide settings and user of that version can see all system messages, because they are in channel in fact.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.0.3
- Operating System: CentOS7
- Deployment Method: docker
- Number of Running Instances: 12
- DB Replicaset Oplog: Enabled
- NodeJS Version: v12.14.0
- MongoDB Version: 4.0.13

### Client Setup Information

- Desktop App Version: 2.17.9
- Mobile App Version: 4.2
- Operating System: Windows 10

**Comment by ankar84** [comment]
Issue is not solved unfortunately.

**Comment by ankar84** [comment]
@ggazzo @rodrigok @sampaiodiego @gabriellsh @mariaeduardacunha 
Can we consider to add optional per channel opportunity to totally disable system messages (like it was before #16243)?
I think that could be the issue with chat list sorted by activity.
If we hide system messages in new way - that system message in fact exist, but hide. 
And as a result all chats configured that way in sort by activity are all on top. 
And when you open it - there is no new messages. 
I think this is because someone joined room or leave it and - that system message is hide, but in fact exist and activity handle it and sort chat to top of the list.

**Comment by rodrigok** [comment]
@ankar84 we discussed this issue recently and decided to make what you suggested now, to only update the room's activity when the system message is visible, so if you configure to hide the system message it will not update the room and it will not move up in your list.

It's planned for the next releases, I just linked this issue to our internal roadmap task.

Thank you for the ping here.

**Comment by ankar84** [comment]
> decided to make what you suggested now, to only update the room's activity when the system message is visible, so if you configure to hide the system message it will not update the room and it will not move up in your list.

Thanks a lot, Rodrigo!
It's a very good news for me and I believe for a lot of users experiencing that kind of issues.
And thanks again, that you and Rocket Chat team are listen to community! That really great!

**Comment by ankar84** [comment]
> It's planned for the next releases, I just linked this issue to our internal roadmap task.

Hi, Rodrigo! @rodrigok 
Version 3.6.0 released recently.
Can you tell when fix for that issue is planned to be implement?
Thank you!

**Comment by ankar84** [comment]
As I see, it looks like it implemented in 3.6.3
My chats not going up in list in web and desktop version anymore after someone join or left that channel.
@rodrigok can you clarify that?
P.S. @diegolmello @djorkaeffalexandre in Mobile version it's not implemented. I will make a separate issue.

**Comment by sampaiodiego** [comment]
I think this was closed by https://github.com/RocketChat/Rocket.Chat/pull/22364

**Comment by ankar84** [comment]
> I think this was closed by #22364

I can check and reopen that issue if it's not.
Thanks, Diego!
